pytorch_lightning/trainer/training_loop.py

```diff
@@ -298,6 +298,12 @@ def training_step(self, split_batch, batch_idx, opt_idx, hiddens):
             args = self.build_train_args(split_batch, batch_idx, opt_idx, hiddens)
             training_step_output = self.trainer.accelerator_backend.training_step(args)
             training_step_output = self.trainer.call_hook('training_step_end', training_step_output)
+            if training_step_output is None:
+                raise MisconfigurationException(
+                    'training_step cannot return None. HINT: Did you add a return statement?'
```

Comment:
Could we add another hint, that the user should use `Trainer.SKIP` if they want to skip the training step?

Reply:
maybe create trainer with state UNUSED?

Reply:
for any of these states, please don’t use decorators... it makes debugging code impossible

Reply:
> Could we add another hint, that the user should use `Trainer.SKIP` if they want to skip the training step?

Sure

> maybe create trainer with state UNUSED?

Mind expanding on what you mean?

Reply:
just thinking what the case here is, I assume that it is about using Trainer.test without fit, right? so to somehow distinguish that training was not used...?

Reply:
It just covers the case of someone returning `None` in `training_step`. Most usual reason is forgetting the `return` statement.

Without that check you just get the following error:

```
  File "python3.8/site-packages/pytorch_lightning/trainer/logging.py", line 140, in process_output
    for k, v in output.items():
AttributeError: 'NoneType' object has no attribute 'items'
```

I don't see what is the relation to `Trainer.test`. If i'm not mistaken, running `test` never calls `training_step`